### ISSUE TITLE
react-a11y-accessible-headings is missing from recommended_ruleset.js

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [Microsoft/tslint-microsoft-contrib#696]
### Bug Report

-   **`tslint-microsoft-contrib` version**: Latest on master

Spinning off of #686 - `react-a11y-accessible-headings` isn't mentioned in `recommended_ruleset.js`. Running `npm run test` re-adds it back in.

**Comment by IllusionMH** [comment]
I have an impression (may be wrong) that contributors are not sure if they should add/commit changes to automatically generated files (specifically `recommended_ruleset.js` and `tslint-warnings.csv`).

It may worth to add information to `CONTRIBUTING.md` that changes in these files related to new rules should be also commited. And during review check if changes are present and ask to add for example `recommended: 'false, // to strict'` to rule metadata.

What do you think?

**Comment by JoshuaKGoldberg** [comment]
Agreed. I think it'd be good to have a comment on top of `recommended_ruleset.js` linking to those docs too.
